mypy/meet.py

```diff
@@ -137,6 +139,24 @@ def get_possible_variants(typ: Type) -> List[Type]:
         return [typ]
 
 
+def is_enum_overlapping_union(x: ProperType, y: ProperType) -> bool:
+    return (
+        isinstance(x, Instance) and x.type.is_enum and
+        isinstance(y, UnionType) and
+        all(isinstance(z, LiteralType) and z.fallback.type == x.type  # type: ignore[misc]
```

Comment:
Why do we need `type: ignore` here? 🤔

Reply:
because when mypy typechecks itself it complains about using isinstance on values of type `Type`, suggesting running them through `get_proper_type` first (which I don't think is appropriate here) or using this `type: ignore`